File: src/main/java/com/thebluehats/server/core/modules/TimerModule.java
package com.thebluehats.server.core.modules;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import com.thebluehats.server.game.managers.enchants.Timer;

import org.bukkit.plugin.java.JavaPlugin;

public class TimerModule extends AbstractModule {
    @Provides
    static Timer<?> provideTimer(JavaPlugin plugin) {
        return new Timer<>(plugin);
    }

    @Override
    protected void configure() {
    }
}

File: src/main/kotlin/com/thebluehats/server/core/modules/TimerModule.kt
package com.thebluehats.server.core.modules

import com.google.inject.AbstractModule
import com.google.inject.Provides
import com.thebluehats.server.game.managers.enchants.Timer
import org.bukkit.plugin.java.JavaPlugin

class TimerModule : AbstractModule() {
    override fun configure() {}

    companion object {
        @Provides
        fun provideTimer(plugin: JavaPlugin): Timer<*> {
            return Timer<Any>(plugin)
        }
    }
}